ci/Jenkinsfile

```diff
@@ -38,7 +38,7 @@ pipeline {
         stage('Builds') {
             failFast true
             parallel {
-                stage('Build:linux:x86_64:gcc') {
+                stage('Build:linux:x86_64:gcc:release') {
```

Comment:
I'd talked about this with David. Based on some bugs we've hit in the past, with strange ODR behavior involving inline function usage differing between debug and release builds (because of the compiler choosing to inline or not), and some other odd behavior arising from interactions with -fvisibility=hidden, its a good idea to build and test both flavors.

Reply:
Thats fair, what if we just made it into a single stage that configures debug mode, builds in debug mode, runs the debug tests and then finally, generates the reports? All in one stage? Then we dont have to have 2 stages of build and test, and we dont need the excess artifacts just to move things between debug build and debug test.